datastream-bigquery/src/main/java/com/linkedin/datastream/bigquery/Batch.java

```diff
@@ -104,20 +102,20 @@ public void write(Package aPackage) throws InterruptedException {
             final GenericRecord record;
             try {
                 record = (GenericRecord) _valueDeserializer.deserialize((byte[]) aPackage.getRecord().getValue());
-            } catch (SerializationException e) {
-                if (e.getCause() instanceof BufferUnderflowException) {
-                    LOG.warn("Error deserializing message at Topic {} - Partition {} - Offset {} - Reason {} - Exception {}",
-                            aPackage.getTopic(),
-                            aPackage.getPartition(),
-                            aPackage.getOffset(),
-                            e.getMessage(),
-                            e.getCause().getClass());
-                    DynamicMetricsManager.getInstance().createOrUpdateMeter(
-                            this.getClass().getSimpleName(),
-                            aPackage.getTopic(),
-                            "deserializerErrorCount",
-                            1);
-                }
+            } catch (Exception e) {
```

Comment:
Which other possible exception you noticed other than SerializationException?
The reason I ask is, if we don't send neg ack where we should have, we see dataloss.

Reply:
The goal of the block here is to capture any exception that is thrown by the `_valueDeserializer.deserialize()`, log the exception with topic/destination information, and record an error count in the metrics. The exception is thrown after logging/measuring, so the the exception handling logic higher up in the stack will handle the retry or re-route to an exceptions table. I changed the scope of the catch block from `SerializationException` to `Exception` to make sure we log/measure other exceptions, i.e. a runtime exception.

Reply:
Never mind, I did not see `throw e;` at the end earlier.